### PR TITLE
Enable the CORS header for AGIC

### PR DESCRIPTION
**Pull request by rachitkumar13**
## Checklist
- [ ] The title of the PR is clear and informative
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] Issues addressed by the PR are mentioned in the description followed by `Fixes`.

### CODE REVIEW

**Comment by akshaysngupta** [comment]
@rachitkumar13 is this PR ready for review ?

**Comment by rachitkumar13** [comment]
> @rachitkumar13 is this PR ready for review ?

Yes

File: docs/annotations.md [hunk 1]
```diff
@@ -850,3 +851,34 @@ spec:
...
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: go-server-ingress-bkprefix
+  namespace: test-ag
+  annotations:
+    kubernetes.io/ingress.class: azure/application-gateway
+    appgw.ingress.kubernetes.io/appgw-cors-enable: true/false
```
Comment (akshaysngupta) [inline docs/annotations.md:875]: How about we shorten the key to "enable-cors"

File: pkg/appgw/requestroutingrules_test.go [hunk 2]
```diff
@@ -935,6 +935,50 @@ var _ = Describe("Test routing rules generations", func() {
...
+		_ = configBuilder.BackendAddressPools(cbCtx)
+		_ = configBuilder.Listeners(cbCtx)
 
+		pathMap := configBuilder.getPathMaps(cbCtx)
+
+		rule := &ingress.Spec.Rules[0]
+		listenerID := generateListenerID(ingress, rule, n.ApplicationGatewayProtocolHTTP, nil, false)
+		var ruleSets *[]n.ApplicationGatewayRewriteRuleSet = configBuilder.appGw.ApplicationGatewayPropertiesFormat.RewriteRuleSets
+		var corsEnabled bool
+		for i := 0; i < len(*ruleSets); i++ {
```
Comment (akshaysngupta) [inline pkg/appgw/requestroutingrules_test.go:964]: This line is throwing nil exception in the test
image

**Review by akshaysngupta — CHANGES_REQUESTED** [review]
Please address the comments

**Pull request closed without merging**